pytorch_lightning/trainer/properties.py

```diff
@@ -483,6 +486,19 @@ def sanity_checking(self, val: bool) -> None:
     Loop properties
     """
 
+    @property
+    def epoch_loop(self) -> EpochLoop:
+        # TODO: the current train_loop should be renamed to epoch_loop
+        return self.train_loop
+
+    @property
+    def training_loop(self) -> TrainingLoop:
+        return self.epoch_loop.training_loop
+
+    @property
+    def batch_loop(self) -> BatchLoop:
+        return self.epoch_loop.training_loop.batch_loop
+
```

Comment:
- why do these need to be exposed?
- with this naming scheme, how do evaluation and predict loops fit in?

Reply:
Those are exposed mainly for convenience, but could also become entry points for customization (we haven't decided on the API there yet)

Reply:
we should probably rename batch_loop to training_batch_loop? The class too.